src/ImageSharp/Formats/Tiff/Writers/TiffBaseColorWriter.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using SixLabors.ImageSharp.Formats.Experimental.Tiff.Compression;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.Metadata.Profiles.Exif;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff.Writers
+{
+    internal abstract class TiffBaseColorWriter<TPixel> : IDisposable
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        private bool isDisposed;
+
+        protected TiffBaseColorWriter(ImageFrame<TPixel> image, MemoryAllocator memoryAllocator, Configuration configuration, TiffEncoderEntriesCollector entriesCollector)
+        {
+            this.Image = image;
+            this.MemoryAllocator = memoryAllocator;
+            this.Configuration = configuration;
+            this.EntriesCollector = entriesCollector;
+
+            this.BytesPerRow = ((image.Width * this.BitsPerPixel) + 7) / 8;
+        }
+
+        public abstract int BitsPerPixel { get; }
+
+        public int BytesPerRow { get; }
+
+        protected ImageFrame<TPixel> Image { get; }
+
+        protected MemoryAllocator MemoryAllocator { get; }
+
+        protected Configuration Configuration { get; }
+
+        protected TiffEncoderEntriesCollector EntriesCollector { get; }
+
+        public virtual void Write(TiffBaseCompressor compressor, int rowsPerStrip)
+        {
+            DebugGuard.IsTrue(this.BytesPerRow == compressor.BytesPerRow || compressor.BytesPerRow == 0, "Values must be equals");
+            int stripsCount = (this.Image.Height + rowsPerStrip - 1) / rowsPerStrip;
+
+            uint[] stripOffsets = new uint[stripsCount];
```

Comment:
Can these be pooled?

Reply:
i dont see how, it depends on the chosen value for BytesPerRow (or maybe i misunderstand you and you mean i just should use the memory allocator here?)

Reply:
Yeah memory allocator

Reply:
Same as below with the colorPalette. I cannot figure out howto do so, because stripOffsets is used further down as a `ExifLongArray` and it expects a byte array there. I cannot use a Span here.

Reply:
No worries, not a priority.